utils.py

```diff
@@ -68,7 +74,10 @@ def query_model_parallelized(model, tokenizer, prompt_list, max_tokens, top_p, t
                     logits_list[i].append(logits)
             final_prompt_hidden_state_list = [None for _ in range(len(prompt_list))]
 
-    generated_answer_list = [s.lower() for s in tokenizer.batch_decode(outputs['sequences'], skip_special_tokens=True)]
+    sequences = outputs['sequences']
+    if tokenizer.chat_template is not None:
```

Comment:
The truncation of chat models is a bit different from regular models and need to be done separately. Also got rid of lowercasing for generation tasks (done below for classification tasks).